addons/repair/models/stock_picking.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+
+
+class PickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    can_repair = fields.Boolean(
+        'Create Repair Orders from Returns',
+        compute='_compute_can_repair', store=True, readonly=False,
+        help="If ticked, you will be able to directly create repair orders from a return.")
+    is_return_type = fields.Boolean(
+        "Technical field to identify default return picking type for a warehouse.", store=True,
+        compute='_compute_can_repair')
+
+    @api.depends('warehouse_id.return_type_id')
+    def _compute_can_repair(self):
+        self.is_return_type = False
+        for picking_type in self:
+            picking_type.is_return_type = picking_type == picking_type.warehouse_id.return_type_id
+            if not picking_type.is_return_type:
+                picking_type.can_repair = False
```

Comment:
I think it's better to check if the picking.type is a return_picking_type_id of another picking.type instead of checking on the warehouse. It will restrict the possible configuration

Reply:
Maybe with a one2many field for the depends trigger

Reply:
This is a tricky thing unfortunately. The feature is specifically only supposed to be for the "Returns" picking type. Since there's no `code=return` option, I originally was inferring the "Returns" from operation types with `code=incoming` + no `return_picking_type_id` to distinguish it from the default "Receipt" type. However the flexibility to create new alternative "Return" types wasn't desired + it also opened the possibility of configuring the types incorrectly and having a weird repair option show up when it shouldn't.

TLDR version: I could make it check for whether a picking type is a `return_picking_type_id` + `code=incoming`, but this is a flexibility that isn't needed/could result in the "repair" option showing up when not desired.

Reply:
Ignore previous comment. This has been changed to only `picking.type.code == 'incoming'` + is a return_picking_type_id of another picking type as suggested.